### ISSUE TITLE
Import of Application Gateway causes panic

### ISSUE THREAD

**Issue by khushil** [terraform-providers/terraform-provider-azurerm#618]
```
terraform@vmcandc:~/lemon$ terraform --version
Terraform v0.11.1
+ provider.azurerm v0.3.3
```

```
terraform@vmcandc:~/lemon$ terraform import azurerm_application_gateway.wafecmpilot /subscriptions/REDACTED/resourceGroups/rgecmpilot/providers/Microsoft.Network/applicationGateways/wafecmpilot

azurerm_application_gateway.wafecmpilot: Importing from ID "/subscriptions/REDACTED/resourceGroups/rgecmpilot/providers/Microsoft.Network/applicationGateways/wafecmpilot"...
azurerm_application_gateway.wafecmpilot: Import complete!
  Imported azurerm_application_gateway (ID: /subscriptions/REDACTED/resourceGroups/rgecmpilot/providers/Microsoft.Network/applicationGateways/wafecmpilot)
azurerm_application_gateway.wafecmpilot: Refreshing state... (ID: /subscriptions/62461cb7-231b-4fcb-9f14-...etwork/applicationGateways/wafecmpilot)
Error importing: 1 error(s) occurred:

* azurerm_application_gateway.wafecmpilot (import id: /subscriptions/REDACTED/resourceGroups/rgecmpilot/providers/Microsoft.Network/applicationGateways/wafecmpilot): 1 error(s) occurred:

* import azurerm_application_gateway.wafecmpilot result: /subscriptions/REDACTED/resourceGroups/rgecmpilot/providers/Microsoft.Network/applicationGateways/wafecmpilot: azurerm_application_gateway.wafecmpilot: unexpected EOF
panic: runtime error: invalid memory address or nil pointer dereference
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11d5f17]
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4:
2017-12-11T22:03:18.212Z [DEBUG] plugin: plugin process exited: path=/home/terraform/lemon/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v0.3.3_x4
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: goroutine 26 [running]:
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.flattenApplicationGatewaySslPolicy(0x0, 0x15f7862, 0x3, 0x15939a0)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_application_gateway.go:1350 +0x37
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmApplicationGatewayRead(0xc42020a380, 0x1543220, 0xc4203a6000, 0x0, 0x1ffdba0)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_application_gateway.go:763 +0x4a6
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc420299320, 0xc4202b82d0, 0x1543220, 0xc4203a6000, 0xc4202e7cb8, 0x432701, 0x8002025ec00)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:321 +0x199
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc42029e850, 0xc4202b8280, 0xc4202b82d0, 0x7f35808646c8, 0x0, 0x18)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:284 +0x9a
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc4202a0ce0, 0xc420478010, 0xc420478320, 0x0, 0x0)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: reflect.Value.call(0xc420055380, 0xc42000c828, 0x13, 0x15f7b3a, 0x4, 0xc42027af20, 0x3, 0x3, 0xc4202b3728, 0x1, ...)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/usr/local/go/src/reflect/value.go:434 +0x906
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: reflect.Value.Call(0xc420055380, 0xc42000c828, 0x13, 0xc4202b3720, 0x3, 0x3, 0x1f93440, 0xc420054780, 0xc4203c6000)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: net/rpc.(*service).call(0xc420210740, 0xc420078eb0, 0xc420250180, 0xc42015c980, 0xc42016e5c0, 0x13aa920, 0xc420478010, 0x16, 0x13aa960, 0xc420478320, ...)
2017-12-11T22:03:18.212Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2017-12-11T22:03:18.213Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: created by net/rpc.(*Server).ServeCodec
2017-12-11T22:03:18.213Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
```

**Comment by SaviMohan** [comment]
I am also experiencing a similar issue when trying to import an azure app service plan.
```
tf --version
Terraform v0.11.1
+ provider.azurerm v0.3.3
```



```
tf import azurerm_app_service_plan.REDACTED/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Web/serverFarms/REDACTED
azurerm_app_service_plan.REDACTED: Importing from ID "/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Web/serverFarms/REDACTED"...
azurerm_app_service_plan.REDACTED: Import complete!
  Imported azurerm_app_service_plan (ID: /subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Web/serverFarms/REDACTED)
azurerm_app_service_plan.REDACTED: Refreshing state... (ID: /subscriptions/REDACTED-.../serverFarms/REDACTED)
Error importing: 1 error(s) occurred:

* azurerm_app_service_plan.base_appserviceplan (import id: /subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Web/serverFarms/REDACTED): 1 error(s) occurred:

* import azurerm_app_service_plan.REDACTED result: /subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Web/serverFarms/REDACTEDpanic: runtime error: invalid memory address or nil pointer dereference
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: [signal 0xc0000005 code=0x0 addr=0x0 pc=0x11f8c81]
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: goroutine 57 [running]:
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmAppServicePlanRead(0xc04219c380, 0x1576e20, 0xc042394000, 0x0, 0x202ce60)
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_app_service_plan.go:168 +0x541
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc042251620, 0xc04215e410, 0x1576e20, 0xc042394000, 0xc04229db20, 0xc04207db01, 0x80000000018)
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:321 +0x1a0
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc0422708c0, 0xc04215e370, 0xc04215e410, 0x1813f8, 0x0, 0x18)
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:284 +0xa1
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc042162de0, 0xc0420323b0, 0xc042032650, 0x0, 0x0)
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x55
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: reflect.Value.call(0xc042332060, 0xc042160628, 0x13, 0x162bf71, 0x4, 0xc042495f20, 0x3, 0x3, 0x0, 0x0, ...)
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /usr/local/go/src/reflect/value.go:434 +0x90d
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: reflect.Value.Call(0xc042332060, 0xc042160628, 0x13, 0xc042495f20, 0x3, 0x3, 0x0, 0x0, 0x0)
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /usr/local/go/src/reflect/value.go:302 +0xab
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: net/rpc.(*service).call(0xc042166c80, 0xc04215e2d0, 0xc04215a490, 0xc04216c780, 0xc042163940, 0x13e0100, 0xc0420323b0, 0x16, 0x13e0140, 0xc042032650, ...)
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /usr/local/go/src/net/rpc/server.go:381 +0x149
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe: created by net/rpc.(*Server).ServeCodec
2017-12-14T10:38:21.915+1300 [DEBUG] plugin.terraform-provider-azurerm_v0.3.3_x4.exe:   /usr/local/go/src/net/rpc/server.go:475 +0x372
2017/12/14 10:38:21 [ERROR] root: eval: *terraform.EvalRefresh, err: azurerm_app_service_plan.REDACTED: unexpected EOF
2017/12/14 10:38:21 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_app_service_plan.REDACTED: unexpected EOF
2017/12/14 10:38:21 [TRACE] [walkImport] Exiting eval tree: import azurerm_app_service_plan.REDACTED result: /subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Web/serverFarms/REDACTED
2017/12/14 10:38:21 [TRACE] dag/walk: upstream errored, not walking "provider.azurerm (close)"
2017/12/14 10:38:21 [DEBUG] plugin: waiting for all plugin processes to complete...
2017-12-14T10:38:21.917+1300 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2017-12-14T10:38:21.933+1300 [DEBUG] plugin: plugin process exited: path=C:\Users\REDACTED\REDACTED\REDACTED\.terraform\plugins\windows_amd64\terraform-provider-azurerm_v0.3.3_x4.exe

: azurerm

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
_app_service_plan.REDACTED: unexpected EOF
```

**Comment by tgeist** [comment]
I too am experiencing this issue with an Azure Application Gateway.  I can import the resource group, the vnet, the subnets, and public IP, but the Application Gateway fails.  It's worth noting that I can import an Application Gateway created with Terraform.  Could this be due to not having all the necessary optional items in my template?  i.e. management certificate, request routing rules, disabled cipher suites, etc?

**Comment by tgeist** [comment]
Just an update - the import will work up until you add backend address pools, health probes, listeners, and routing rules.

**Comment by jcleavel** [comment]
I am having an almost identical error, but with `azurerm_traffic_manager_profile` while attempting a `terraform import`.

```
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 2018/01/23 17:53:30 [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Network/trafficmanagerprofiles/REDACTED?api-version=2017-05-01:
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: HTTP/1.1 200 OK
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: Transfer-Encoding: chunked
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: Cache-Control: private
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: Content-Type: application/json; charset=utf-8
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: Date: Wed, 24 Jan 2018 01:53:29 GMT
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: Server: Microsoft-IIS/8.5
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: Strict-Transport-Security: max-age=31536000; includeSubDomains
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: Vary: Accept-Encoding
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: X-Aspnet-Version: 4.0.30319
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: X-Content-Type-Options: nosniff
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: X-Ms-Correlation-Request-Id: f4f5e62c-28ad-4d56-839a-920f79a0760e
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 10798
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: X-Ms-Request-Id: fe78dae9-2322-4309-b4e1-cad10e5ba60f
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: X-Ms-Routing-Request-Id: CANADAEAST:20180124T015330Z:f4f5e62c-28ad-4d56-839a-920f79a0760e
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: X-Powered-By: ASP.NET
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 5bb
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 0
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 
2018-01-23T17:53:30.396-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: panic: runtime error: invalid memory address or nil pointer dereference
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: [signal 0xc0000005 code=0x0 addr=0x0 pc=0x139484b]
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: goroutine 85 [running]:
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/azurerm.flattenAzureRMTrafficManagerProfileMonitorConfig(...)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_traffic_manager_profile.go:269
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmTrafficManagerProfileRead(0xc0421ca620, 0x1647ca0, 0xc0423be000, 0x0, 0x2106080)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_traffic_manager_profile.go:183 +0x99b
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc04231ad20, 0xc04216a1e0, 0x1647ca0, 0xc0423be000, 0xc042342e58, 0xc042332a01, 0x80000000018)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:321 +0x1a0
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc042152930, 0xc04216a190, 0xc04216a1e0, 0x43306a8, 0x0, 0x18)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:284 +0xa1
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc04214db20, 0xc0422ac5f0, 0xc0422ac7a0, 0x0, 0x0)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x55
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: reflect.Value.call(0xc042352360, 0xc042074b28, 0x13, 0x16d40ba, 0x4, 0xc0424aff20, 0x3, 0x3, 0x0, 0x0, ...)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/usr/local/go/src/reflect/value.go:434 +0x90c
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: reflect.Value.Call(0xc042352360, 0xc042074b28, 0x13, 0xc0424aff20, 0x3, 0x3, 0x0, 0x0, 0x0)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/usr/local/go/src/reflect/value.go:302 +0xab
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: net/rpc.(*service).call(0xc042053e80, 0xc04208c6e0, 0xc042056a70, 0xc04215c780, 0xc04216e5a0, 0x147f340, 0xc0422ac5f0, 0x16, 0x147f380, 0xc0422ac7a0, ...)
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/usr/local/go/src/net/rpc/server.go:381 +0x149
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: created by net/rpc.(*Server).ServeCodec
2018-01-23T17:53:30.399-0800 [DEBUG] plugin.terraform-provider-azurerm_v1.0.1_x4.exe: 	/usr/local/go/src/net/rpc/server.go:475 +0x372
2018/01/23 17:53:30 [ERROR] root.common_module: eval: *terraform.EvalRefresh, err: azurerm_traffic_manager_profile.tm: unexpected EOF
2018/01/23 17:53:30 [ERROR] root.common_module: eval: *terraform.EvalSequence, err: azurerm_traffic_manager_profile.tm: unexpected EOF
2018/01/23 17:53:30 [TRACE] [walkImport] Exiting eval tree: import module.common_module.azurerm_traffic_manager_profile.tm result: /subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Network/trafficManagerProfiles/REDACTED
2018/01/23 17:53:30 [TRACE] dag/walk: upstream errored, not walking "provider.azurerm (close)"
2018/01/23 17:53:30 [TRACE] dag/walk: upstream errored, not walking "root"
2018/01/23 17:53:30 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-01-23T17:53:30.404-0800 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2018-01-23T17:53:30.408-0800 [WARN ] plugin: error closing client during Kill: err="unexpected EOF"
2018-01-23T17:53:30.412-0800 [DEBUG] plugin: plugin process exited: path=C:\Users\jcleavel\projects\PATH_REDACTED\.terraform\plugins\windows_amd64\terraform-provider-azurerm_v1.0.1_x4.exe
2018-01-23T17:53:30.415-0800 [DEBUG] plugin: plugin process exited: path=C:\Users\jcleavel\projects\PATH_REDACTED\.terraform\plugins\windows_amd64\terraform-provider-null_v1.0.0_x4.exe
```

I can provide the entire log if requested.

**Comment by tombuildsstuff** [comment]
@jcleavel from what I can see I believe I've fixed the crash you're seeing yesterday and will ship as part of v1.0.2 (it's currently merged into master, if you're interested in using a custom build in the meantime to workaround it).

Thanks!

**Comment by jcleavel** [comment]
Hey, that's great! Thanks @tombuildsstuff!

**Comment by tombuildsstuff** [comment]
👋 hey all

From what I can see this was closed in v1.0.2 - as such I'm going to close this issue, however please feel free to reopen it if this is still an issue and we'll take another look :)

Thanks!
